C_enums (#1909)
Codecademy/docs: content/c/concepts/enums/enums.md

> # Enum

Reviewer: This entry is missing the required metadata. To understand what to put here, please read the [metadata section in the style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#metadata). The metadata should look like this:

```
---
Title: 'The Title'
Description: 'A brief description'
Subjects:
  - 'A subject name'
  - 'A second subject name'
  - 'An nth subject name'
Tags:
  - 'A tag'
  - 'A second tag'
  - 'An nth tag'
CatalogContent:
  - 'learn-example-course'
  - 'path/example-path'
---

Also, we do not have h1 headers to title Doc entries, the title in the metadata does this automatically.

A **enum** (enumeration) is a user-defined data type in C. It is used to assign names to integral constants, the principal purpose of this names is to create a program that is easy to read and maintain. It is defined using the `enum` keyword.

## Syntax

```pseudo
enum name
{
	memberName1 = <integral constant>,
	memberName2,
	memberName3,
	...
	memberNameN
};
```

An enum is made up of members where each `memberName` can have a defined integral constant (i.e. 0, 10, 2) or just the `memberName`.

## Example

The following example uses the `enum` keyword to create a `day` enumeration:

```c
enum day {
	Sunday, 
	Monday,
	Tuesday,
	Wednesday,
	Thursday,
	Friday,
	Saturday
};
```

`Sunday` will be setvalue 0, `Monday` will be set to a value of 1, and so  `Patient` structure.

> ## Another Example

Reviewer: You don't need these two examples. The second example below illustrates the same thing. I'd remove these first two, and move that second example below so it's the first example.

```c

// Another example program to demonstrate working
// of enum in C
#include<stdio.h>
 
enum year{Jan, Feb, Mar, Apr, May, Jun, Jul,
          Aug, Sep, Oct, Nov, Dec};
 
int main()
{
   int i;
   for (i=Jan; i<=Dec; i++)   
      printf("%d ", i);
   
   return 0;
}
```

### Output

```c
0 1 2 3 4 5 6 7 8 9 10 11
```

> ## Note

Reviewer: ```suggestion
## Examples 
```

> ### 1. Two enum names can have the same value. For instance, both ‘Running’ and ‘In_Progress’ have the same value of 1.

Reviewer: ```suggestion
Two `enum` names can have the same value. For instance, both `Running` and `In_Progress` have the same value of 1.
```
These are not appropriate as headers.

```c
#include <stdio.h>
enum State {Todo = 0, Running = 1, In_Progress = 1};
int main()
{
    printf("%d, %d, %d", Todo, Running, In_Progress);
    return 0;
}
```

> #### Output

Reviewer: ```suggestion
This example results in the following output.
```

```shell
0, 1, 1
```

> ### 2. If we do not explicitly assign values to enum MemberName, the compiler by default assigns values starting from 0. For instance, Sunday gets a value of 0, Monday gets 1, and so on.

Reviewer: ```suggestion
If we do not explicitly assign values to `enum` `day`, the compiler by default assigns values starting from 0. For instance, `sunday` gets a value of 0, `monday` gets 1, and so on.
```

Can we also move this so it is the first example?

```c
#include <stdio.h>
enum day {sunday, monday, tuesday, wednesday, thursday, friday, saturday};
 
int main()
{
    enum day d = wednesday;
    printf("The day number stored in d is %d", d);
    return 0;
}
```

> #### Output

Reviewer: ```suggestion
This example results in the following output:
```

```shell
The day number stored in d is 3
```

> ### 3. We can assign values to a particular name in any order. All unassigned names get value as the value of the previous name plus one.

Reviewer: ```suggestion
We can assign values to a particular name in any order. All unassigned names get value as the value of the previous name plus one.
```

```c
#include <stdio.h>
enum day {sunday = 1, monday, tuesday = 20,
          wednesday, thursday = 10, friday, saturday};
 
int main()
{
    printf("%d %d %d %d %d %d %d", sunday, monday, tuesday,
            wednesday, thursday, friday, saturday);
    return 0;
}
```

> #### Output

Reviewer: ```suggestion
This example results in the following output:
```

```shell
1 2 20 21 10 11 12
```

> ### 4.  All enum constants must be unique in their scope.

Reviewer: ```suggestion
> **Note:**  All `enum` constants must be unique in their scope.
```

---
Title: 'Enum'

> Description: 'Enum is used to assign custom names to integral constants'

Reviewer: ```suggestion
Description: 'An enum is used to assign custom names to integral constants.'
```

Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:

>   - 'Enums'

Reviewer: ```suggestion
  - 'Enum'
```

>   - 'Variable Type'

Reviewer: ```suggestion
  - 'Variable Types'
```

CatalogContent:
  - 'learn-c'
  - 'paths/computer-science'
---

> A **enum** (enumeration) is a user-defined data type in C. It is used to assign names to integral constants, the principal purpose of this names is to create a program that is easy to read and maintain. It is defined using the `enum` keyword.

Reviewer: ```suggestion
An **enum** (enumeration) is a user-defined data type in C. It is used to assign names to integral constants. The principal purpose of these names is to create a program that is easy to read and maintain. It is defined using the `enum` keyword.
```

## Syntax

```pseudo
enum name
{

> 	memberName1 = <integral constant>,

Reviewer: ```suggestion
  memberName1 = <integral constant>,
```

> 	memberName2,

Reviewer: ```suggestion
  memberName2,
```

> 	memberName3,

Reviewer: ```suggestion
  memberName3,
```

> 	...

Reviewer: ```suggestion
  ...
```

> 	memberNameN

Reviewer: ```suggestion
  memberNameN
```

};
```

> An enum is made up of members where each `memberName` can have a defined integral constant (i.e. 0, 10, 2) or just the `memberName`.

Reviewer 2: ```suggestion
An enum is made up of members where each `memberName` can have a defined integral constant (e.g. 0, 10, 2) or just the `memberName`.
```

## Examples

> If we do not explicitly assign values to `enum` `day`, the compiler by default assigns values starting from 0. For instance, `sundays` gets a value of 0, `monday` gets 1, and so on.

Reviewer: ```suggestion
If we do not explicitly assign values to `enum` `day`, the compiler by default assigns values starting from 0. For instance, `sunday` gets a value of 0, `monday` gets 1, and so on.
```
